File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.cpp
class Solution {
public:
    int findPairs(vector<int>& nums, int k) {
        int i = 0;
        int j = 1;
        sort(nums.begin(), nums.end());

        set<pair<int, int>> ans;

        while(j < nums.size()) {
            int diff = nums[j] - nums[i];

            if(diff == k) {
                ans.insert({nums[i], nums[j]});
                i++;
                j++;
            } else if(diff < k) {
                j++;
            } else {
                i++;
            }

            if(i == j) j++;
        }

        return ans.size();
    }
};

File: 0532-k-diff-pairs-in-an-array/0532-k-diff-pairs-in-an-array.java
class Solution {
    public int findPairs(int[] nums, int k) {
        int count = 0;
        Arrays.sort(nums);

        for(int i = 0; i<nums.length-1; i++) {
            if(i > 0 && nums[i] == nums[i-1]) continue;
            int j = i+1;

            while(j < nums.length) {
                if(Math.abs(nums[i] - nums[j]) == k) {
                    count++;
                }
                j++;
                while(j < nums.length && nums[j] == nums[j-1]) j++;
            }
        }

        return count;
    }
}